nrf-softdevice/src/events.rs

```diff
@@ -15,7 +15,7 @@ static SWI2_WAKER: AtomicWaker = AtomicWaker::new();
 #[repr(u32)]
 #[derive(Debug, PartialEq, Eq, Clone, Copy, IntoPrimitive, TryFromPrimitive)]
 #[cfg_attr(feature = "defmt", derive(defmt::Format))]
-enum SocEvent {
+pub enum SocEvent {
     Hfclkstarted = raw::NRF_SOC_EVTS_NRF_EVT_HFCLKSTARTED,
     PowerFailureWarning = raw::NRF_SOC_EVTS_NRF_EVT_POWER_FAILURE_WARNING,
     FlashOperationSuccess = raw::NRF_SOC_EVTS_NRF_EVT_FLASH_OPERATION_SUCCESS,
```

Comment:
Now that is enum is public, it shouldn't have the `FlashOperationSuccess`, `FlashOperationError` variants since these are never returned to the user. Them being there might make the user think they have to handle these events themselves when using the flash API, or make them confused if they try to handle them and never receive them.